pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -327,3 +334,79 @@ func getKubernetesResources(kogitoRes *kogitores.KogitoAppResources) []utilsres.
 
 	return k8sRes
 }
+
+// Ensure that all Kogito images are available before the build.
+func (r *ReconcileKogitoApp) ensureKogitoImageStream(instance *v1alpha1.KogitoApp) (reconcile.Result, error) {
+	isISPresentOnOpenshiftNamespace := true
+
+	imageStreamTag := kogitores.ImageStreamTag
+	if instance.Spec.Build.ImageS2I.ImageStreamTag != "" {
+		imageStreamTag = instance.Spec.Build.ImageS2I.ImageStreamTag
+		log.Infof("Using ImageS2I Tag from CR: %s", imageStreamTag)
+	}
+
+	// First look on the current namespace, if the imagestreams are available, return
+	// and don't check for anything else.
+	for _, key := range kogitores.ImageStreamNameList {
+		is, _ := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: key, Namespace: instance.Namespace}, imageStreamTag)
+		if is == nil {
+			r.isImagestreamReady = false
+		}
+	}
+
+	// look for the Kogito image stream on the default namespace (openshift)
+	// All images must be present, if not create it on current namespace
+	if !r.isImagestreamReady {
+		for _, key := range kogitores.ImageStreamNameList {
+			isOcpNs, _ := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: key, Namespace: kogitores.ImageStreamNamespace}, imageStreamTag)
+			if isOcpNs == nil {
+				isISPresentOnOpenshiftNamespace = false
+			}
+		}
+	}
+
+	if !isISPresentOnOpenshiftNamespace && !r.isImagestreamReady {
+
+		for _, is := range kogitores.KogitoImageStream(instance.Namespace, imageStreamTag).Items {
+			is.Namespace = instance.Namespace
+			hasIs, err := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: is.Name, Namespace: instance.Namespace}, imageStreamTag)
+			if err != nil {
+				log.Error(err.Error())
+			}
+
+			if hasIs == nil {
+				r.isImagestreamReady = false
+				_, err := openshift.ImageStreamC(r.client).CreateImageStream(&is)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+
+				// wait for 5 seconds to the image get ready
```

Comment:
Instead of waiting for the image stream to create, we usually create the resources and  requeue the reconcile loop for a period of time, like this [line](https://github.com/kiegroup/kogito-cloud-operator/blob/31d5dc655e1ab86bf42a5b328b4a8ecda1356ddd/pkg/controller/kogitoapp/kogitoapp_controller.go#L251) . The reconcile loop will be triggered again after the specified period of time.

Reply:
Just read your comment after writing mine :(

Reply:
+1
I liked this idea, now this is the behaviour.